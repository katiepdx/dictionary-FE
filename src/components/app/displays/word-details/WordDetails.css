.label {
  background-color: rgba(216, 179, 136, 0.294);
}

.details {
  display: grid;
  grid-template-columns: 5vw 1fr 5vw;
}
