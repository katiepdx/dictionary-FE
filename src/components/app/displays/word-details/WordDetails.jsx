import React from 'react'
import PropTypes from 'prop-types'
import styles from './WordDetails.css'
import { Link } from 'react-router-dom'

const WordDetails = ({
  id,
  word,
  wordLanguage,
  wordTranslation,
  wordDefinition,
  exampleSentence,
  notes }) => {
  return (
    <div>
      <h1>{word}</h1>
      <p className={styles.label}>Language</p>
      <p> {wordLanguage}</p>

      <p className={styles.label}>Translation</p>
      <p> {wordTranslation ? wordTranslation : 'n/a'}</p>

      <p className={styles.label}>Definition</p>
      <p> {wordDefinition}</p>

      <p className={styles.label}>Example Sentence</p>
      <p> {exampleSentence}</p>

      <p className={styles.label}>Notes</p>
      <p> {notes ? notes : 'n/a'}</p>

      <Link to={`/words/update/${id}`}><button>Update</button></Link>
      <Link to={`/words/delete/${id}`}><button>Delete</button></Link>
    </div>
  )
}

WordDetails.propTypes = {
  id: PropTypes.number.isRequired,
  word: PropTypes.string.isRequired,
  wordLanguage: PropTypes.string.isRequired,
  wordTranslation: PropTypes.string,
  wordDefinition: PropTypes.string.isRequired,
  exampleSentence: PropTypes.string.isRequired,
  notes: PropTypes.string
}

export default WordDetails
