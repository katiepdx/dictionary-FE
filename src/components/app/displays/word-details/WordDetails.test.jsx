import React from 'react'
import { render } from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom'
import WordDetails from './WordDetails'

describe('<WordDetails/>', () => {
  it('should render the word details to the page', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <WordDetails
          id={5}
          word="WordDetailTest"
          wordLanguage="WordDetailTest"
          wordTranslation="WordDetailTest"
          wordDefinition="WordDetailTest"
          exampleSentence="WordDetailTest"
          notes="I learned a new word!"
        /></MemoryRouter>)

    expect(asFragment()).toMatchSnapshot()
  })
})
