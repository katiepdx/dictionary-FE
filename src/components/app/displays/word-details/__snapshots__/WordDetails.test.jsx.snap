// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WordDetails/> should render the word details to the page 1`] = `
<DocumentFragment>
  <div
    class="details"
  >
    <div />
    <div>
      <h1>
        WordDetailTest
      </h1>
      <p
        class="label"
      >
        Language
      </p>
      <p>
         WordDetailTest
      </p>
      <p
        class="label"
      >
        Translation
      </p>
      <p>
         WordDetailTest
      </p>
      <p
        class="label"
      >
        Definition
      </p>
      <p>
         WordDetailTest
      </p>
      <p
        class="label"
      >
        Example Sentence
      </p>
      <p>
         WordDetailTest
      </p>
      <p
        class="label"
      >
        Notes
      </p>
      <p>
         I learned a new word!
      </p>
      <a
        href="/words/update/5"
      >
        <button>
          Update
        </button>
      </a>
      <a
        href="/words/delete/5"
      >
        <button>
          Delete
        </button>
      </a>
    </div>
    <div />
  </div>
</DocumentFragment>
`;
